subprojects/core-api/src/main/java/org/gradle/api/provider/ListProperty.java

```diff
@@ -18,16 +18,47 @@
 
 import org.gradle.api.Incubating;
 
+import javax.annotation.Nullable;
+import java.util.Collection;
 import java.util.List;
 
 /**
  * Represents a property whose type is a {@link List} of elements of type {@link T}.
  *
- * <p><b>Note:</b> This interface is not intended for implementation by build script or plugin authors. An instance of this class can be created through the factory method {@link org.gradle.api.provider.ProviderFactory#listProperty(Class)}.
+ * <p><b>Note:</b> This interface is not intended for implementation by build script or plugin authors. An instance of
+ * this class can be created through the factory method {@link org.gradle.api.provider.ProviderFactory#listProperty(Class)}.
  *
  * @param <T> the type of elements.
  * @since 4.3
  */
 @Incubating
 public interface ListProperty<T> extends PropertyState<List<T>> {
+    /**
+     * Adds an element to the property without evaluating existing providers already present.
+     *
+     * @param element The element, can be null.
```

Comment:
what happens when the element is null? what happens to the existing value of this property? what happens if this property had no value?

Reply:
Good catch, I forgot to update the documentation. A `NullPointerException` will be thrown if `element` is null. Same as the other add method, the existing value of the property will be kept. If the property has no value, adding will succeed but it will throw an IllegalStateException when querying the base provider as it has no values.